components/layout_thread/lib.rs

```diff
@@ -858,6 +869,9 @@ impl LayoutThread {
         // Drop the root flow explicitly to avoid holding style data, such as
         // rule nodes.  The `Stylist` checks when it is dropped that all rule
         // nodes have been GCed, so we want drop anyone who holds them first.
+        let now = std_time::precise_time_ns();
+        let waiting_time = now - self.layout_query_timestamp;
+        debug!("layout: query has been waited: {}", waiting_time);
```

Comment:
This is the wrong time to measure this value, since this only happens when a page is closed.